apps/condo/domains/miniapp/components/GlobalApps/GlobalAppsFeaturesContext.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { createContext, useCallback, useState, useContext } from 'react'
+import pick from 'lodash/pick'
+import { EventEmitter } from 'eventemitter3'
+import { B2BAppGlobalFeature, Scalars } from '@app/condo/schema'
+
+// Specify all data needed for specific feature
+type MapGenerationFeatureContext = {
+    feature: B2BAppGlobalFeature.PropertyMapGeneration,
+    propertyId: Scalars['ID']
+}
+
+// Group all features contexts using |
+export type FeatureContext = MapGenerationFeatureContext
+
+// Store all miniapps available features
+type IFeaturesType = { [key in B2BAppGlobalFeature]?: string }
+// Register new features
+type IRegisterFeaturesType = (newFeatures: IFeaturesType) => void
+// Request app to launch a feature
+type IRequestFeatureAction = (context: FeatureContext) => void
+// Handle feature launch request
+export type IRequestFeatureHandler = (context: FeatureContext) => void
+
+type IGlobalAppsFeaturesContext = {
+    features: IFeaturesType
+    registerFeatures: IRegisterFeaturesType
+    requestFeature: IRequestFeatureAction
+    addFeatureHandler: (handler: IRequestFeatureHandler) => void,
+    removeFeatureHandler: (handler: IRequestFeatureHandler) => void,
+}
+
+const GlobalAppsFeaturesContext = createContext<IGlobalAppsFeaturesContext>({
+    features: {},
+    registerFeatures: () => ({}),
```

Comment:
May this types been specified to correspond with `FeaturesEmitter` object declared bellow?

Reply:
That's not type, it's default implementation for case, when there's no Context provider. (For example in CC)